app/pages/project/index.jsx

```diff
@@ -0,0 +1,428 @@
+    this.setState({
+      error: null,
+      loading: true,
+      preferences: null
+    });
+
+    const slug = `${ownerName}/${projectName}`;
+
+    return apiClient.type('projects').get({ slug, include: 'avatar,background,owners' })
+      .then(([project]) => {
+        this.setState({ project });
+
+        if (project != null) {
+          // Use apiClient with cached resources from include to get out of cache
+          let awaitOrganization;
+          const awaitBackground = apiClient.type('backgrounds').get(project.links.background.id)
+          .catch(error => {
+            if (error.status === 404) { return { src: '' }; } else { return console.error(error); }
+          });
+
+          if ((project.links != null ? project.links.organization : undefined) != null) {
+            awaitOrganization = project.get('organization', { listed: true })
+              .catch(error => [])
+              .then(response => (response != null ? response.display_name : undefined) ? response : null);
+          } else {
+            awaitOrganization = Promise.resolve(null);
+          }
+
+          const awaitOwner = apiClient.type('users').get(project.links.owner.id).catch(error => console.error(error));
+
+          const awaitPages = project.get('pages').catch(error => []); // does not appear in project links?
+
+          const awaitProjectAvatar = apiClient.type('avatars').get(project.links.avatar.id).catch(error => null);
+
+          const awaitProjectCompleteness = Promise.resolve(project.completeness === 1.0);
+
+          const awaitProjectRoles = getAllLinked(project, 'project_roles').catch(error => console.error(error));
+
+          const awaitPreferences = this.getUserProjectPreferences(project, user);
+
+          const awaitTranslation = this.props.actions.translations.load('project', project.id, this.props.translations.locale);
+
+          Promise.all([
+            awaitBackground,
+            awaitOrganization,
+            awaitOwner,
+            awaitPages,
+            awaitProjectAvatar,
+            awaitProjectCompleteness,
+            awaitProjectRoles,
+            awaitPreferences,
+            awaitTranslation
+          ])
+          .then(([
+            background,
+            organization,
+            owner,
+            pages,
+            projectAvatar,
+            projectIsComplete,
+            projectRoles,
+            preferences
+          ]) => {
+            const ready = true;
+            this.setState({ background, organization, owner, pages, projectAvatar, projectIsComplete, projectRoles, preferences, ready });
+            this.loadFieldGuide(project.id);
+            this.props.actions.translations.loadTranslations('project_page', pages.map(page => page.id), this.props.translations.locale);
+          })
+          .catch((error) => {
+            this.setState({ error });
+            console.error(error);
+          });
+        } else {
+          this.setState({
```

Comment:
Out of scope of a conversion PR, but does the code ever get to this else? Wouldn't a 404 be caught by the catch of the Promise?

Reply:
Beats me, but the render method does check for the `NOT_FOUND` error message which implies the code does end up here if the project doesn't exist.

I think this might be using an API route that returns an array of projects, rather than a specific project, in which case it won't 404.